finesse/hardware/device.py

```diff
@@ -170,25 +170,16 @@ class Device(AbstractDevice):
     defined as device base types or not.
     """
 
-    def __init_subclass__(cls, is_base_type: bool = False, **kwargs: Any) -> None:
-        """Initialise a device type class.
-
-        Args:
-            is_base_type: Whether this class represents a device base type
-        """
-        # If it is a base class, we initialise it as such
-        if is_base_type:
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        """Initialise a device type class."""
+        if _base_types.isdisjoint(cls.__mro__):
```

Comment:
I've tried to figure out what this `__mro__` attribute is doing but could not figure out from the obscure explanations I've found. Could you give a one-liner explanation of what this is for?

Reply:
It contains *all* of the ancestor classes of `cls`. Before, we were just looking at the class's immediate parents, but now we check if any of the ancestors is a device base type.

Reply:
(I found the explanations online confusing too :laughing:)